artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/queue/FederatedQueue.java

```diff
@@ -63,7 +63,7 @@ public FederatedQueue(Federation federation, FederationQueuePolicyConfiguration
       Objects.requireNonNull(config.getName());
       this.config = config;
       this.priorityAdjustment = federationUpstream.getPriorityAdjustment() + (config.getPriorityAdjustment() == null ? -1 : config.getPriorityAdjustment());
-      String metaDataFilterString = config.isIncludeFederated() ? null : FederatedQueueConsumer.FEDERATION_NAME +  " IS NOT NULL";
+      String metaDataFilterString = config.isIncludeFederated() ? null : "hyphenated_props:" + FederatedQueueConsumer.FEDERATION_NAME +  " IS NOT NULL";
```

Comment:
It is meant to be circular in a directional setup, if theres a local consumer messages wont go back but should a message move to a broker and then consumer go then message should return back to first broker if there is one there

Reply:
If you want to restrict that then you should use hops to limit it as per docs

Reply:
-1 as this would break the original intent.